src/validation.cpp

```diff
@@ -2472,6 +2472,8 @@ void CChainState::PruneBlockIndexCandidates() {
 /**
  * Try to make some progress towards making pindexMostWork the active block.
  * pblock is either nullptr or a pointer to a CBlock corresponding to pindexMostWork.
+ *
+ * @returns true unless a system error occurred
  */
 bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainParams& chainparams, CBlockIndex* pindexMostWork, const std::shared_ptr<const CBlock>& pblock, bool& fInvalidFound, ConnectTrace& connectTrace)
```

Comment:
As the method is defined there, shouldn't this documentation be in the `.h` file? (same for `CChainState::ActivateBestChain` below)

Reply:
`ActivateBestChainStep` is marked `private` and was never meant to be exposed in the header. I think it makes sense to keep the documentation close to the function implementation

Reply:
Good point for the `public` method, though. So I fixed this for ABC only